src/components/dashboard/main-content/course-enrollments/CourseEnrollments.jsx

```diff
@@ -53,12 +57,12 @@ const CourseEnrollments = ({ children }) => {
     );
     setAssignments(assignmentsData);
 
-    const hasCancelledAssignments = assignmentsData?.some(
-      assignment => assignment.state === ASSIGNMENT_TYPES.CANCELLED,
-    );
-    const hasExpiredAssignments = assignmentsData?.some(assignment => isAssignmentExpired(assignment));
+    const hasActiveCancelledAssignments = assignmentsData?.some((assignment) => (
+      assignment.state === ASSIGNMENT_TYPES.CANCELLED)) && getIsActiveCancelledAssignment(assignmentsData);
+    setShowCancelledAssignmentsAlert(hasActiveCancelledAssignments);
 
-    setShowCancelledAssignmentsAlert(hasCancelledAssignments);
+    const hasExpiredAssignments = assignmentsData?.some(assignment => isAssignmentExpired(assignment))
+      && getIsActiveExpiredAssignment(assignmentsData);
```

Comment:
Related to another comment, I believe you could move the call to the (refactored) `isAssignmentExpired` into `getIsActiveExpiredAssignment` itself such that it has access to the proposed `enrollByDeadline` returned by `isAssignmentExpired` to compare against the localStorage timestamp value.